### ISSUE TITLE
Follow-Up Features:

### ISSUE THREAD

**Issue by laurentsenta** [laurentsenta/pl-diagnose#13]

**Comment by laurentsenta** [comment]
Another thing is filtering out private IP addresses from the part that gives back providers

**Comment by laurentsenta** [comment]
Yes. One of the main challenges with ipns is figuring out which records are found on the DHT
It’s currently also hard to know if you’re using a gateway, whether the problem is IPNS or the underlying CID

A classic example for which we don’t have any tooling is
https://discuss.ipfs.tech/t/ipns-name-resolve-fails-the-first-time-name-publish-and-name-resolve-are-called-but-works-on-next-call/15523/5

Having some endpoint where I can give an ipns name and it returns for example the ipns record and count of the different versions of the record it finds on the DHT would be useful
Basically between an ipns name and a resolution there are all sorts of possible states that are invisible with all current tooling
Another thing is filtering out private IP addresses from the part that gives back providers

**Comment by laurentsenta** [comment]
https://filecoinproject.slack.com/archives/C03FPDYG6T0/p1677592636879719?thread_ts=1677573662.966089&cid=C03FPDYG6T0
